tensorflow/lite/micro/cortex_m_generic/micro_time.cc

```diff
@@ -31,6 +35,18 @@ namespace tflite {
 uint32_t ticks_per_second() { return 0; }
 uint32_t GetCurrentTimeTicks() { return 0; }
 
+#elif defined(USE_SEMIHOSTING)
+
+// For QEMU Semihosting these reqeusts are passed to the host OS.
+uint32_t ticks_per_second() { return 0; }
+
+uint32_t GetCurrentTimeTicks() {
+  uint32_t ms = std::chrono::duration_cast<std::chrono::milliseconds>(
```

Comment:
@mbernico Thank you so much for this PR! I gave it a go and noticed that this line always returns the same (non-zero) number each run. This is not the case on x86. Any idea what the issue is?

Reply:
Very good question.  I'm not much of a C++ developer yet, but here is what I think is happening...  

In the makefile, line 229 I'm using semihosting with ARM (PLATFORM_FLAGS += --specs=rdimon.specs)

When you do that, it stubs out system calls that would need to be defined if we were compiling for a specific board, and generally passes them to the host operating system when QEMU is used.  This allows me to run the tests in user space, really quickly, but it's not a perfect test because we aren't building against a specific device.

To that end, I think that std::chrono::system_clock::now() probably makes a call to a system call that isn't built out.  It's just a stub.  See this for a discussion on the same. https://community.renesas.com/mcu-mpu/embedded-system-platform/f/forum/6267/gettimeofday-always-returns-17-for-the-time

It sounds like alternatively I could have implemented my own version of all those system calls but I'm not even close to smart enough to do that. So, instead I set the ticks to 0 so that test would pass.  :)

Reply:
let's do this in smaller chunks. For now we are going with a separate target `cortex_m_qemu` so we will end up using `micro/micro_time.cc` which defaults to returning 0.

We can investigate other ways to perhaps get meaninful profiling information from qemu later.

Reply:
Okay, SGTM